### PR TITLE
Fix CVE issues

### PR DESCRIPTION
**Pull request by res-life**
### Problem
Should fix the CVE issues in the docker image.

### Changes
- Update Alluxio to 2.9.0
- Fix CVE issues by workaround method: we build Alluxio ourselves, exclude and upgrade some 3PP jars.

Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

File: Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+set -e
+
+# clone, apply patches and compile
```
Comment (tgravescs) [inline Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh]: add more description as to what this is doing and why.

File: Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+cd /tmp/Alluxio-2.9.0-without-CVE
+git config --global user.email "you@example.com"
+git config --global user.name "Your Name"
+git clone https://github.com/Alluxio/alluxio.git
+cd alluxio/
+git checkout v2.9.0 -b v2.9.0-fix-cve
+git am < /tmp/Alluxio-2.9.0-without-CVE/patches/0001-Fix-CVE-issues.patch
+git am < /tmp/Alluxio-2.9.0-without-CVE/patches/0002-Fix-CVE-issues.patch
```
Comment (tgravescs) [inline Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh]: can we just have this apply any patches in that directory and do it based on number order?

File: Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
```
Comment (tgravescs) [inline Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh:1]: needs NVIDIA copyright, that applies to all files

File: Databricks/Alluxio-2.9.0-without-CVE/libexec/alluxio-config.sh [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+# available at www.apache.org/licenses/LICENSE-2.0
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+
+# Included in all the Alluxio scripts with source command should not be executable directly also
+# should not be passed any arguments, since we need original $*
```
Comment (tgravescs) [inline Databricks/Alluxio-2.9.0-without-CVE/libexec/alluxio-config.sh]: why is this file here?  description of changes.

File: Databricks/Alluxio-2.9.0-without-CVE/patches/readme.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+The patch files is from this PR: https://github.com/res-life/alluxio/pull/1
```
Comment (tgravescs) [inline Databricks/Alluxio-2.9.0-without-CVE/patches/readme.txt]: we dont' need this if patch files in this pr.

File: Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh (hunk 1 above)
Comment (res-life) [reply Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh]: Done

File: Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh (hunk 2 above)
Comment (res-life) [reply Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh]: Done

File: Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh (hunk 3 above)
Comment (res-life) [reply Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh:1]: Done

File: Databricks/Alluxio-2.9.0-without-CVE/libexec/alluxio-config.sh (hunk 4 above)
Comment (res-life) [reply Databricks/Alluxio-2.9.0-without-CVE/libexec/alluxio-config.sh]: See the comments in the `build-from-src-and-fix-CVE.sh`
```
# update `libexec/alluxio-config.sh`, replace the client and server jar as the Alluxio building script does.
# For details, refer to:
# https://github.com/Alluxio/alluxio/blob/v2.9.0/dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go#L274-L275
git am /tmp/Alluxio-2.9.0-without-CVE/update-assembly-jar-path.patch
```

File: Databricks/Alluxio-2.9.0-without-CVE/patches/readme.txt (hunk 5 above)
Comment (res-life) [reply Databricks/Alluxio-2.9.0-without-CVE/patches/readme.txt]: Removed this `readme` file.

**Comment by res-life** [comment]
Done.
Please review it again.

File: Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+echo "Compiling Alluxio, this may take some time ..."
+mvn clean install -q -Pufs-hadoop-3 -Dufs.hadoop.version=3.2.4 -Dmaven.javadoc.skip=true -Dmaven.test.skip=true -Dlicense.skip=true -Dcheckstyle.skip=true -Dfindbugs.skip=true -Dhadoop.version=3.2.4 -T 4
+echo "Compile done"
+
+# setup Alluxio program located in `/opt` path
+cd ../
+mkdir alluxio-2.9.0
+cp -r ./alluxio/bin ./alluxio-2.9.0
+mkdir ./alluxio-2.9.0/assembly
+cp ./alluxio/assembly/server/target/alluxio-assembly-server-2.9.0-jar-with-dependencies.jar ./alluxio-2.9.0/assembly/alluxio-server-2.9.0.jar
```
Comment (tgravescs) [inline Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh:47]: this seems odd why are we renaming?

File: Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+# update `libexec/alluxio-config.sh`, replace the client and server jar as the Alluxio building script does.
+# For details, refer to:
+# https://github.com/Alluxio/alluxio/blob/v2.9.0/dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go#L274-L275
+git am /tmp/Alluxio-2.9.0-without-CVE/update-assembly-jar-path.patch
+
+echo "Compiling Alluxio, this may take some time ..."
+mvn clean install -q -Pufs-hadoop-3 -Dufs.hadoop.version=3.2.4 -Dmaven.javadoc.skip=true -Dmaven.test.skip=true -Dlicense.skip=true -Dcheckstyle.skip=true -Dfindbugs.skip=true -Dhadoop.version=3.2.4 -T 4
+echo "Compile done"
+
+# setup Alluxio program located in `/opt` path
```
Comment (tgravescs) [inline Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh:41]: so this section is copying only specific jars so we can avoid CVE issues in the unused jars, correct?  If so please add comment

File: Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh (hunk 6 above)
Comment (res-life) [reply Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh:47]: It's what the Alluxio packaging script does.

File: Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh (hunk 7 above)
Comment (res-life) [reply Databricks/Alluxio-2.9.0-without-CVE/build-from-src-and-fix-CVE.sh:41]: Done.

**Review by tgravescs — APPROVED** [review]
